numpy/lib/_datasource.py

```diff
@@ -333,12 +332,9 @@ def _cache(self, path):
 
         # TODO: Doesn't handle compressed files!
         if self._isurl(path):
-            try:
-                with closing(urlopen(path)) as openedurl:
-                    with _open(upath, 'wb') as f:
-                        shutil.copyfileobj(openedurl, f)
-            except URLError:
-                raise URLError("URL not found: %s" % path)
+            openedurl = urlopen(path)
```

Comment:
The `with` statement is still needed, even if `closing` is not.

Reply:
I'm sorry :sweat: , is it fine now?